### PR TITLE
[ZEPPELIN-3311] Zeppelin should ask user to re-login after JWT cookie expires (KnoxJwtRealm)

### PR DESCRIPTION
**Pull request by prabhjyotsingh**
### What is this PR for?
Zeppelin should ask user to re-login after JWT cookie expires.

### What type of PR is it?
[Bug Fix]

### What is the Jira issue?
* [ZEPPELIN-3311](https://issues.apache.org/jira/browse/ZEPPELIN-3311)

### How should this be tested?
Steps to reproduce :
1) Enable SSO for zeppelin
2) Login to zeppelin using SSO.
3) Now wait till JWT expires(knoxsso.token.ttl under Advanced knoxsso-topology could be set to lower values to get a short lived cookie)
4) Refresh zeppelin. Zeppelin doesn't ask for re-login, though JWT is already expired and not valid anymore.

### Questions:
* Does the licenses files need update? N/A
* Is there breaking changes for older versions? N/A
* Does this needs documentation? N/A

### CODE REVIEW

File: zeppelin-server/src/main/java/org/apache/zeppelin/realm/jwt/KnoxJwtRealm.java [hunk 1]
```diff
@@ -182,6 +193,33 @@ protected boolean validateSignature(SignedJWT jwtToken) {
...
+   * subclasses in order to customize the expiration validation behavior.
+   *
+   * @param jwtToken
+   *            the token that contains the expiration date to validate
+   * @return valid true if the token has not expired; false otherwise
+   */
+  protected boolean validateExpiration(SignedJWT jwtToken) {
+    boolean valid = false;
+    try {
+      Date expires = jwtToken.getJWTClaimsSet().getExpirationTime();
```
Comment (zjffdu) [inline zeppelin-server/src/main/java/org/apache/zeppelin/realm/jwt/KnoxJwtRealm.java]: Does LOCALE matter here ?

Comment (prabhjyotsingh) [reply zeppelin-server/src/main/java/org/apache/zeppelin/realm/jwt/KnoxJwtRealm.java]: It's taken from https://github.com/apache/ranger/blob/master/security-admin/src/main/java/org/apache/ranger/security/web/filter/RangerSSOAuthenticationFilter.java#L467-L491.
ExpirationTime is passed as an epoch, so it should not matter.

Comment (prabhjyotsingh) [reply zeppelin-server/src/main/java/org/apache/zeppelin/realm/jwt/KnoxJwtRealm.java]: Yes, that I understood LOCALE for this context is time zone, yes it should not matter.

**Comment by r-kamath** [comment]
LGTM

File: zeppelin-server/src/main/java/org/apache/zeppelin/realm/jwt/KnoxJwtRealm.java (hunk 1 above)
Comment (zjffdu) [reply zeppelin-server/src/main/java/org/apache/zeppelin/realm/jwt/KnoxJwtRealm.java]: Thanks, LGTM

**Comment by prabhjyotsingh** [comment]
Thanks for the review will merge this if no more discussion.

**Pull request closed without merging**